library/std/src/io/readbuf.rs

```diff
@@ -58,6 +60,12 @@ impl<'a> ReadBuf<'a> {
         ReadBuf { buf, filled: 0, initialized: 0 }
     }
 
+    /// Creates a new [`ReadBufRef`] referencing this `ReadBuf`.
+    #[inline]
+    pub fn borrow(&mut self) -> ReadBufRef<'_, 'a> {
+        ReadBufRef { read_buf: self }
+    }
+
```

Comment:
Should this implement the [`Borrow` trait](https://doc.rust-lang.org/std/borrow/trait.Borrow.html) instead?

Reply:
I don't think it can. The borrow trait returns `&Borrowed` which not work for this API

Reply:
If the name was changed to `ReadBufGuard` or similar as suggested above, both `borrow()` and `reborrow()` methods could be renamed to `guard()`?

Reply:
I think `borrow` is the right name, following `RefCell`. We only use `guard` in the context of an RAII guard and this isn't following that pattern.